eb-api/src/main/java/io/engineblock/activityapi/rates/AverageRateLimiter.java

```diff
@@ -129,36 +128,20 @@ protected long getNanoClockTime() {
     @Override
     public long acquire(long nanos) {
 
-        long opScheduleTime = ticksTimeline.getAndAdd(nanos);
-        long timelinePosition = lastSeenNanoTime;
-
-        if (opScheduleTime < timelinePosition) {
-            isBursting = true;
-            return reportCoDelay ? timelinePosition - opScheduleTime : 0L;
+        long opScheduleTimeNs = ticksTimeline.getAndAdd(nanos);
+        long delayNs = opScheduleTimeNs - getNanoClockTime();
```

Comment:
I believe that calling getNanoClockTime() every time here is not necessary. Given that the clock read is cached in the atomic, it is possible for subsequent callers to see that their scheduled time chunk already fits within 'at least' some cached value. This means that advancing the cached value is only necessary as an exception to this condition. The maximal rate limiter performance varies significantly between these two implementations (I've tested this extensively), and I'd like to keep it as low-overhead as possible.

On the other hand, if you can make a compelling argument that it is indeed incorrect, it should absolutely be changed to be correct.

Reply:
I've done a rewrite of the rate limiter. Should I close out this PR pending your input on the new implementation?